### ISSUE TITLE
TypeError: Cannot read property 'interval_ms' of undefined

### ISSUE THREAD

**Issue by kempsteven** [tvjsx/trading-vue-js#225]
## Description
I am trying to update the an overlays data, specifically "ROC" Overlay, i am using `.merge` to update the data,
i use that to update chart.data `this.chart.merge('chart.data', data)`, and it works perfectly, but when I update an Overlay like
ROC I get the error `TypeError: Cannot read property 'interval_ms' of undefined`

## Screenshot 
![image](https://user-images.githubusercontent.com/51051704/114704476-3adacc80-9d59-11eb-9619-d4303c7fd3a0.png)

## Code (MVP)
``` js
const data = [[1617978600000, 2.38], [1618237800000, 2.31], [1618324200000, 1.58]]
this.chart.merge(`offchart.ROC5C.data`, data)
```

## Console errors (if any)
![image](https://user-images.githubusercontent.com/51051704/114704476-3adacc80-9d59-11eb-9619-d4303c7fd3a0.png)
## Additional Info 

(tvjs version, OS, browser, npm version...)
"trading-vue-js": "^1.0.2"

**Comment by kempsteven** [comment]
I've looked through the source code and it seems `this.$refs.chart` & `this.tv.$refs.chart` is undefined, not sure why tho

**Comment by mjzarrin** [comment]
I had a similar problem and the problem was about the frequency of calling update().
by adding some sleep between calls, the problem was solved.

Here is my working example .
```
async mounted() {
    for (const cdl of json.chart.data) {
      await new Promise((resolve) => setTimeout(resolve, 100))
      this.chart.update({ candle: cdl })
    }
  },
```

**Comment by oneart-dev** [comment]
Same problem. Do you find solution @kempsteven ?

**Comment by oneart-dev** [comment]
@mjzarrin where do you put this code? I have this problem when adding a new indicator? not on initial load.

**Comment by Swoorup** [comment]
Proper fix is to give the timeframe interval when you specify the datacube. 

```javascript
this.chart = new DataCube({
   ohlcv: data['chart.data'],
   chart: { tf: '1m '},
```

**Comment by oneart-dev** [comment]
@Swoorup nope, the error not about missing TF but not able to get chart ref because its undefined. I found my problem.
I was using v-if on TradingView component while loading chart data. Removing it fixed the issue.

So the main problem is tv extensions do not check for undefined refs.
